api/v1alpha3/awsmachine_types.go

```diff
@@ -73,12 +73,12 @@ type AWSMachineSpec struct {
 	// FailureDomain is the failure domain unique identifier this Machine should be attached to, as defined in Cluster API.
 	// For this infrastructure provider, the ID is equivalent to an AWS Availability Zone.
 	// If multiple subnets are matched for the availability zone, the first one return is picked.
-	FailureDomain *string `json:"failureDomainID,omitempty"`
+	FailureDomain *string `json:"failureDomain,omitempty"`
 
 	// AvailabilityZone is references the AWS availability zone to use for this instance.
 	// If multiple subnets are matched for the availability zone, the first one return is picked.
 	//
-	// DEPRECATED: Switch to FailureDomainID.
+	// DEPRECATED: Switch to FailureDomain.
```

Comment:
Not directly related to this PR, but since we are deprecating this field, should we be handling some type of automatic migration to the FailureDomain field as part of the conversion webhook? It looks like we are doing the reverse (populating the AvailabilityZone in v1alpha2 if FailureDomain is set in v1alpha3).

Reply:
Yeah we can, we might want to remove the field altogether though, this is only marked as a deprecation and we have a fallback in place that looks at the `AvailabilityZone` field if FailureDomain is nil.

Reply:
I'm good with removing the field altogether as well :)

Reply:
I'll file an issue so we don't forget as this is out of scope for the original intent of this PR